From 1bab0b51542a3c79aca2d1b7f6f5cc86590fbb5a Mon Sep 17 00:00:00 2001
From: Benno Schulenberg <bensberg@justemail.net>
Date: Sun, 28 Feb 2016 12:10:02 +0100
Subject: [PATCH 5/5] Don't continue when something is wrong -- skip the rest of the line.

(This is revision 5691 from SVN.  It avoids a segfault on faulty lines.)

Signed-off-by: Benno Schulenberg <bensberg@justemail.net>
---
 src/rcfile.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/src/rcfile.c b/src/rcfile.c
index 785cd90..b190b52 100644
--- a/src/rcfile.c
+++ b/src/rcfile.c
@@ -879,8 +879,7 @@ void parse_header_exp(char *ptr)
 	if (*ptr != '"') {
 	    rcfile_error(
 		N_("Regex strings must begin and end with a \" character"));
-	    ptr = parse_next_regex(ptr);
-	    continue;
+	    return;
 	}
 
 	ptr++;
-- 
1.7.0.4

